tools/run_tests/build_python.sh

```diff
@@ -51,11 +52,25 @@ fi
 
 tox -e ${TOX_PYTHON_ENV} --notest
 
+# We force the .so naming convention in PEP 3149 for side by side installation support
+# Note this is the default in Python3, but explicitly disabled for Darwin, so we only
+# use this hack for our testing environment.
+if [ "$PY_VERSION" -gt "27" ]
+then
+  mv $ROOT/src/python/grpcio/grpc/_cython/cygrpc.so $ROOT/src/python/grpcio/grpc/_cython/cygrpc.so.backup || true
```

Comment:
This *should* get obviated by #6791, I think (because the extension will only be present in the relevant virtual environments, not just sitting in the source directory where things can get confused).